x-pack/plugins/ml/public/application/model_management/test_models/models/index_input.tsx

```diff
@@ -147,6 +182,61 @@ export function useIndexInput({
   const [fieldNames, setFieldNames] = useState<Array<{ value: string; text: string }>>([]);
   const selectedField = useObservable(inferrer.getInputField$(), inferrer.getInputField());
 
+  const supportedFieldTypes = useMemo(() => inferrer.getSupportedFieldTypes(), [inferrer]);
+
+  const fetchCompanitbleIndices = useCallback(async () => {
+    try {
+      const indicesResponse = await mlApiServices.getIndices();
+
+      const indices: string[] = indicesResponse
+        .filter((index) => !index.name.startsWith('.') && !index.hidden)
+        .map((index) => index.name);
+
+      const { indices: compatibleIndices } = await mlApiServices.getFieldCaps({
+        index: indices.join(','),
+        types: supportedFieldTypes,
+        filters: ['-metadata'],
+        includeEmptyFields: false,
+      });
+
+      const adHocDataViews = (
+        await Promise.all(
+          (compatibleIndices as string[]).map(async (index) => {
+            const dataView = await dataViews.create({
+              title: index,
+              name: index,
+            });
+            return dataView;
+          })
+        )
+      )
+        // Fields caps response include irrelevant indcies, hence need to filter
+        .filter((v) => containsSupportedFields(v, supportedFieldTypes));
+
+      if (adHocDataViews.length === 0) {
+        return;
+      }
+
+      setDataViewListItems((prev) => {
+        return [
+          ...prev,
+          {
+            isGroupLabelOption: true,
+            label: i18n.translate('xpack.ml.trainedModels.testModelsFlyout.adHocDataViews', {
+              defaultMessage: 'Created from compatible indices',
```

Comment:
Is there a need to allow users to run against a wildcard index pattern? e.g. test against `partial-.ds-cluster-kibana-filebeat-*` here:

<img width="585" alt="Screenshot 2024-04-15 at 14 44 11" src="https://github.com/elastic/kibana/assets/7405507/cedc7784-66c1-4ef1-b14a-8e501a7398a6">

Reply:
Now possible with ability to create an adhoc data view matching a wildcard pattern